orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pollers/TemporaryServerGroupsPoller.java

```diff
@@ -0,0 +1,244 @@
+import com.netflix.spinnaker.orca.pipeline.ExecutionLauncher;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import groovy.util.logging.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import redis.clients.jedis.Jedis;
+import redis.clients.util.Pool;
+import rx.Observable;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.TemporaryServerGroupEntityTagGenerator.TTL_TAG;
+import static java.lang.String.format;
+
+@Slf4j
+@Component
+@ConditionalOnExpression(value = "${pollers.temporaryServerGroups.enabled:false}")
+public class TemporaryServerGroupsPoller extends AbstractPollingNotificationAgent {
+  private static final Logger log = LoggerFactory.getLogger(TemporaryServerGroupsPoller.class);
+
+  private final ObjectMapper objectMapper;
+  private final OortService oortService;
+  private final RetrySupport retrySupport;
+  private final ExecutionLauncher executionLauncher;
+
+  private final PollerSupport pollerSupport;
+
+  private final Counter errorsCounter;
+  private final Counter triggeredCounter;
+
+  @Autowired
+  public TemporaryServerGroupsPoller(@Qualifier("jedisPool") Pool<Jedis> jedisPool,
+                                     ObjectMapper objectMapper,
+                                     OortService oortService,
+                                     RetrySupport retrySupport,
+                                     Registry registry,
+                                     ExecutionLauncher executionLauncher) {
+    super(jedisPool);
+
+    this.objectMapper = objectMapper;
+    this.oortService = oortService;
+    this.retrySupport = retrySupport;
+    this.executionLauncher = executionLauncher;
+
+    this.pollerSupport = new PollerSupport(objectMapper, retrySupport, oortService);
+
+    this.errorsCounter = registry.counter("poller.temporaryServerGroups.errors");
+    this.triggeredCounter = registry.counter("poller.temporaryServerGroups.triggered");
+  }
+
+  @Override
+  protected long getPollingInterval() {
+    return TimeUnit.MINUTES.toSeconds(10);
+  }
+
+  @Override
+  protected String getNotificationType() {
+    return "temporaryServerGroups";
+  }
+
+  @Override
+  protected void startPolling() {
+    subscription = Observable
+      .timer(getPollingInterval(), TimeUnit.SECONDS, scheduler)
+      .repeat()
+      .filter(interval -> tryAcquireLock())
```

Comment:
Just following the new pattern -- should acquire a lock for the interval.

Reply:
Might just be me, but I prefer the explicit lock vs. message on queue. Easier for me to reason about.